### PR TITLE
Fix clippy warnings about using if-let over match

### PR DESCRIPTION
**Pull request by lberrymage**
The specific lint triggered was [clippy::single_match](https://rust-lang.github.io/rust-clippy/master/index.html#single_match).

**Description**
Details found in the link above

**Testing**
N/A

### CODE REVIEW

File: wgpu-core/src/command/bundle.rs [hunk 1]
```diff
@@ -931,19 +931,16 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
...
-                let bundle = &bundle_guard[id];
-                let label = desc.label.as_ref().map(|l| l.as_ref());
-                trace.lock().add(trace::Action::CreateRenderBundle {
-                    id: id.0,
-                    desc: trace::new_render_bundle_encoder_descriptor(label, &bundle.context),
-                    base: BasePass::from_ref(bundle.base.as_ref()),
-                });
-            }
-            None => {}
+        if let Some(ref trace) = device.trace {
```
Comment (kvark) [inline wgpu-core/src/command/bundle.rs:934]: This has a higher MSRV (minimum supported Rust version), which is the reason it's not written this way originally.
Could you check what the exact MSRV is required with this change?

Comment (lberrymage) [reply wgpu-core/src/command/bundle.rs:934]: Oh whoops, I didn't realize that. MSRV for this is 1.43.0.

Comment (kvark) [reply wgpu-core/src/command/bundle.rs:934]: Hmm, it looks to be [matching Firefox 78](https://wiki.mozilla.org/Rust_Update_Policy_for_Firefox), so we should should be good.

**Pull request merged**